math/utils_test.go

```diff
@@ -283,6 +370,127 @@ func TestWeightedInferences2(t *testing.T) {
+		alloraMath.MustNewDecFromString("0.2"),
+		alloraMath.MustNewDecFromString("0.3"),
+		alloraMath.MustNewDecFromString("0.4"),
+		alloraMath.MustNewDecFromString("0.5"),
 	}
+	percentiles := []alloraMath.Dec{
+		alloraMath.MustNewDecFromString("10"),
+		alloraMath.MustNewDecFromString("25"),
+		alloraMath.MustNewDecFromString("50"),
+		alloraMath.MustNewDecFromString("75"),
+		alloraMath.MustNewDecFromString("90"),
+	}
+	_, err := alloraMath.WeightedPercentile(data, weights, percentiles)
+	require.Error(t, err)
+	require.ErrorIs(t, err, alloraMath.ErrNaN)
+
+	data = []alloraMath.Dec{
+		alloraMath.MustNewDecFromString("10"),
+		alloraMath.MustNewDecFromString("20"),
+		alloraMath.MustNewDecFromString("30"),
+		alloraMath.MustNewDecFromString("40"),
+		alloraMath.MustNewDecFromString("50"),
+	}
+	weights = []alloraMath.Dec{
+		alloraMath.MustNewDecFromString("0.1"),
+		alloraMath.MustNewDecFromString("0.2"),
+		alloraMath.NewNaN(),
+		alloraMath.MustNewDecFromString("0.4"),
+		alloraMath.MustNewDecFromString("0.5"),
+	}
+	_, err = alloraMath.WeightedPercentile(data, weights, percentiles)
+	require.Error(t, err)
+	require.ErrorIs(t, err, alloraMath.ErrNaN)
+
+	weights = []alloraMath.Dec{
+		alloraMath.MustNewDecFromString("0.1"),
+		alloraMath.MustNewDecFromString("0.2"),
+		alloraMath.MustNewDecFromString("0.3"),
+		alloraMath.MustNewDecFromString("0.4"),
+		alloraMath.MustNewDecFromString("0.5"),
+	}
+	percentiles = []alloraMath.Dec{
+		alloraMath.MustNewDecFromString("10"),
+		alloraMath.MustNewDecFromString("25"),
+		alloraMath.MustNewDecFromString("50"),
+		alloraMath.NewNaN(),
+		alloraMath.MustNewDecFromString("90"),
+	}
+	_, err = alloraMath.WeightedPercentile(data, weights, percentiles)
+	require.Error(t, err)
+	require.ErrorIs(t, err, alloraMath.ErrNaN)
+}
+
+func TestGetSortedKeys(t *testing.T) {
+	m := map[int32]struct{}{
+		5: {},
+		3: {},
+		2: {},
+		4: {},
+		1: {},
+	}
+
+	expected := []int32{1, 2, 3, 4, 5}
+	keys := alloraMath.GetSortedKeys[int32](m)
+	require.Equal(t, expected, keys)
+
+	m2 := map[string]struct{}{
+		"5.5":  {},
+		"3.25": {},
+		"2.1":  {},
+		"4.75": {},
+		"1":    {},
+	}
+	expected2 := []string{
+		"1",
+		"2.1",
+		"3.25",
+		"4.75",
+		"5.5",
+	}
+	keys2 := alloraMath.GetSortedKeys(m2)
+	require.Equal(t, expected2, keys2)
+}
+
+func TestGetSortedKeysWithEmptyMap(t *testing.T) {
+	m := map[int32]struct{}{}
+	keys := alloraMath.GetSortedKeys(m)
+	require.Equal(t, []int32{}, keys)
+}
+
+func TestGetSortedElementsByDecWeightDesc(t *testing.T) {
+	dec1 := alloraMath.MustNewDecFromString("0.5")
+	dec2 := alloraMath.MustNewDecFromString("0.2")
+	dec3 := alloraMath.MustNewDecFromString("0.1")
+	dec4 := alloraMath.MustNewDecFromString("0.7")
+	dec5 := alloraMath.MustNewDecFromString("0.4")
+	m := map[int32]*alloraMath.Dec{
+		1: &dec1,
+		2: &dec2,
+		3: &dec3,
+		4: &dec4,
+		5: &dec5,
+	}
+
+	expected := []int32{4, 1, 5, 2, 3}
+	keys := alloraMath.GetSortedElementsByDecWeightDesc(m)
```

Comment:
Some of this unit tests do not include `NaN`s. 
IAs (Copilot too) do a good job at generating batteries of test cases (incl edge cases) for each function including numbers like "NaN" or "+/-Infinity" and providing expected results.

Reply:
yeah this test specifically is not meant to test NaNs, it's a happy path test

I didnt' include NaN tests for this function because the function itself doesn't check for NaNs

which I think is ok, because it's a sort function - it's not processing the number, let it sort the NaN